Samples/HelloWorld/OrleansClient/Program.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Threading;
+using HelloWorld.Interfaces;
+using Orleans;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+
+namespace OrleansClient
+{
+    /// <summary>
+    /// Orleans test silo client
+    /// </summary>
+    public class Program
+    {
+        static int Main(string[] args)
+        {
+            var config = ClientConfiguration.LocalhostSilo();
+            const int initializeAttemptsBeforeFailing = 5;
+            int attemp = 0;
+            while (true)
+            {
+                try
+                {
+                    GrainClient.Initialize(config);
+                    Console.WriteLine("Client successfully connect to silo host");
+                    break;
+                }
+                catch (SiloUnavailableException e)
+                {
+                    attemp++;
+                    if (attemp > initializeAttemptsBeforeFailing)
+                    {
+                        Console.WriteLine("Connect to silo host failed due to " + e);
+                        Console.WriteLine("Press Enter to terminate...");
+                        Console.ReadLine();
+                        return 1;
+                    }
+                    Thread.Sleep(TimeSpan.FromSeconds(2));
+                }
+            }
+
+            var friend = GrainClient.GrainFactory.GetGrain<IHello>(0);
+            Console.WriteLine("\n\n{0}\n\n", friend.SayHello("Good morning, my friend!").Result);
```

Comment:
I understand this is similar to the other hello world, but I always disliked that it confuses people into thinking they must use `.Result` after every grain call from the client.
I'd move this to its own async method such as:
```
private async Task DoClientWork()
{
      var friend = GrainClient.GrainFactory.GetGrain<IHello>(0);
      var response = await friend.SayHello("Good morning, my friend!")
      Console.WriteLine("\n\n{0}\n\n", response);
} 
```
and then just call it from here by using `DoClientWork.Wait()`. It accomplishes a very similar thing, it's just that it's clear that you can (and should) await grain calls. The .Wait or .Result is just a side effect of having the Main entry point not being asynchronous, but not of Orleans programming model, and for real world apps, you will never have all of your client code embedded in the Program.Main method.